app/controllers/user/shares_controller.rb

```diff
@@ -33,27 +33,24 @@ def create
     @share = create_share_with_auditing
 
     if @share.valid?
-      respond_with(@share, location: user_birth_record_path(@share.document))
+      respond_with(@share, location: user_document_path(@share.document.document_type, @share.document.id))
     else
       respond_with(@share)
     end
   end
 
   def revoke
     @share.revoke(revoked_by: current_user)
-    respond_with(@share, location: user_birth_record_path(@share.document))
+    respond_with(@share, location: user_document_path(@share.document.document_type, @share.document.id))
   end
 
   private
 
   def create_share_with_auditing
-    birth_record = current_user.birth_records.find_by(id: share_params['document_id'])
-    authorize birth_record, :share?
-    recipient = Organisation.find_by(id: share_params['recipient_id'])
-    birth_record.share_with(
-      recipient: recipient,
-      user: current_user
-    )
+    document = current_user.documents(type: share_params[:document_type])
+                           .find_by(id: share_params[:document_id])
+    authorize document, :share?
```

Comment:
i wonder if this should be `authorize @share, :create?` instead.
the pundit would check that you're sharing a document that is yours.

Reply:
it's nice when controllers are straight forward
```
@share = Share.new(share_params)
authorize @show
@share.save!
```

Reply:
(but then the auditing... i've usualy put that on the model `after_save`)

Reply:
have updated that now